### PR TITLE
Fix incompleteness and continue with send guard

### PR DESCRIPTION
**Pull request by jcp19**
This PR adapts the code for the send guard to side-step a completeness issue with Gobra/silicon. I made a few simplifications as well, and adapted the code accordingly. Currently, I am having severe perf. issues in this PR that @marcoeilers is looking into. After this is solved, I plan to merge this PR back in #263

### CODE REVIEW

**Comment by jcp19** [comment]
After this is merged, it may be worth looking at #277

**Comment by jcp19** [comment]
This depends on adding support for https://github.com/viperproject/silicon/pull/821 in Gobra and using artificial triggers inside the IO spec

**Review by mlimbeck — APPROVED** [review]

**Review by Dspil — APPROVED** [review]

File: router/io-spec.gobra [hunk 1]
```diff
@@ -646,16 +648,25 @@ pure func (d *DataPlane) dpSpecWellConfiguredLinkTypes(dp io.DataPlaneSpec) bool
...
 	return unfolding acc(d.Mem(), _) in
 		d.dpSpecWellConfiguredLocalIA(dp)     &&
 		d.dpSpecWellConfiguredNeighborIAs(dp) &&
 		d.dpSpecWellConfiguredLinkTypes(dp)
 }
 
+ghost
+requires dp.Valid()
+requires acc(msg.Mem(), R50)
+decreases
```
[the remark this answers is not in the document]
Comment (jcp19) [reply router/io-spec.gobra:662]: Hmm, I am not so sure about it. It contains an unfold of a not too large predicate and a call to a pure function that is itself opaque. I am not sure if, here, the payoff of not using automation is worth it. Specially, when I add the annotation for `moreJoins`

**Comment by jcp19** [comment]
This has the same errors as the basis IO PR. I will merge it now.

**Pull request merged**